java/arcs/sdk/testing/TestingHandles.kt

```diff
@@ -0,0 +1,132 @@
+package arcs.sdk.testing
+
+import arcs.sdk.BaseParticle
+import arcs.sdk.Entity
+import arcs.sdk.ReadWriteCollectionHandle
+import arcs.sdk.ReadWriteSingletonHandle
+
+/**
+ * Self containing implementation of [ReadWriteCollectionHandle] to be used in Unit Tests.
+ */
+class TestingCollection<T : Entity>(
```

Comment:
I have a feeling it might be easier to fake the storage proxy through a TestHandlerManager, and reuse most of the existing classes. The Handle classes are changing a lot and I have a feeling it'll be a pain to maintain both. 

Couldn't we just use a VolatileStorage and the real storage stack? I was imagining JUnit4 ClassRule  that would setup and drive particle lifecycle without needing an ArcHost/Allocator and using RamDisk/Volatile storage.

If these handles don't have CRDT backed storageproxies, don't we run the risk that the tests diverge from real behavior too much?

Reply:
I agree. If it's possible to do, I think using the existing handle implementations backed by RamDisk or Volatile would be better. There would be multiple benefits:
1. We can exercise the real implementations some more.
2. Behaviour in tests is more likely to be the same as in production.
3. Less duplicated code (which is particularly handy given that the handle implementations are still under so much flux lately)

I could even see this working by driving an ArcHost. Just running an arc containing a single particle with its input and output handles.

Reply:
Just to comment without biasing either way, I think the suggestions would (1) make these more like integration tests vs. unit tests and (2) require they be MediumTest due to (likely, presumably) use of threads. The suggestions could still certainly make sense from an optimal-value-added, defer to thoughts from others.

Reply:
Ok, I've reworked stuff to use the real storage stack, PTAL.